src/source-nfq.c

```diff
@@ -504,7 +504,11 @@ static void NFQReleasePacket(Packet *p)
 static int NFQBypassCallback(Packet *p)
 {
     if (IS_TUNNEL_PKT(p)) {
-        SCMutex *m = p->root ? &p->root->tunnel_mutex : &p->tunnel_mutex;
+        if (p->root == NULL) {
```

Comment:
This breaks the tunnel logic. The idea is that IS_TUNNEL_PKT returns true if this packet is either the root packet or the encapsulated packet. In case it's the root p->root == NULL (see IS_TUNNEL_ROOT_PKT), if p->root != NULL it's the inner packet. The mark should always be set on the root packet of course, as that is the only one getting verdicted by nfq.